### PR TITLE
feat: adding step 'pods converge to selector' + fix: CI & syntax

### PR DESCRIPTION
**Pull request by garomonegro**
- adds new step implemented by `PodsInNamespaceWithLabelSelectorConvergeToFieldSelector`
- fixes syntax generation broken by the new step
  - adds `replace` package to better handle the replacement of brackets in the syntax generation.
- fixes CI broken by codecov
- splits `CI / make all` into `CI / build-test-coverage` & `CI / syntax` for clarity
- renames `golangci-lint / lint` to `CI / golangci-lint` for consistancy
- adds `codecov.yml`

### CODE REVIEW

File: docs/syntax.md [hunk 1]
```diff
@@ -34,6 +34,7 @@ Below you will find the step syntax next to the name of the method it utilizes.
 - `<GK> some pods in namespace <non-whitespace-characters> with selector <non-whitespace-characters> don't have "<any-characters-except-(")>" in logs since <any-characters-except-(")> time` kdt.KubeClientSet.SomePodsInNamespaceWithSelectorDontHaveStringInLogsSinceTime
 - `<GK> [the] pods in namespace <non-whitespace-characters> with selector <non-whitespace-characters> have no errors in logs since <any-characters-except-(")> time` kdt.KubeClientSet.PodsInNamespaceWithSelectorHaveNoErrorsInLogsSinceTime
 - `<GK> [the] pods in namespace <non-whitespace-characters> with selector <non-whitespace-characters> have some errors in logs since <any-characters-except-(")> time` kdt.KubeClientSet.PodsInNamespaceWithSelectorHaveSomeErrorsInLogsSinceTime
+- `<GK> [all] [the] (pod|pods) in [the] namespace <non-whitespace-characters> with [the] label selector <non-whitespace-characters> [should] converge to [the] field selector <non-whitespace-characters>` kdt.KubeClientSet.PodsInNamespaceWithLabelSelectorConvergeToFieldSelector
```
Comment (danielhelfand) [inline docs/syntax.md:37]: What does converge mean in this context?

File: pkg/kube/pod/pod_test.go [hunk 2]
```diff
@@ -131,3 +133,61 @@ func Test_PodsInNamespaceWithSelectorShouldHaveLabels(t *testing.T) {
...
+			Phase: v1.PodSucceeded,
+		},
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "Positive Test",
```
Comment (danielhelfand) [inline pkg/kube/pod/pod_test.go:165]: Should we add a case where some pods with the label have the field selector set to succeeded and some do not? Should that result in an error in that case.

Comment (garomonegro) [reply pkg/kube/pod/pod_test.go:165]: I tried implementing that before marking the PR as ready for review. Listing pods with the fake client returns all the pods added to it, regardless if they have the label-selector/field-selector

File: docs/syntax.md (hunk 1 above)
Comment (garomonegro) [reply docs/syntax.md]: Now it can be "converge to" or "have".

**Review by danielhelfand — APPROVED** [review]

**Review by kevdowney — APPROVED** [review]
LGTM

**Pull request merged**